File: src/lib/software/docker_desktop.py
from src.lib.software import BaseSoftware

class DockerDesktop(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'Docker Desktop Installer.exe'

    def resolve_download_url(self):
        self.url_resolved.emit('https://desktop.docker.com/win/main/amd64/Docker%20Desktop%20Installer.exe')

File: packages/app/src/services/software/definitions/dockerDesktop.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'docker-desktop';
	public name = 'Docker Desktop';
	public category = [SoftwareCategory.Development];
	public downloadName = 'Docker Desktop Installer.exe';
	public icon = 'docker-desktop.png';
	public homepage = 'https://docker.com/products/docker-desktop';

	public async resolveDownloadUrl() {
		return ok('https://desktop.docker.com/win/main/amd64/Docker%20Desktop%20Installer.exe');
	}
}
